processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1783,9 +1798,13 @@ private BeanMappingMethod(Method method,
         this.propertyMappings = propertyMappings;
         this.returnTypeBuilder = returnTypeBuilder;
         this.finalizerMethod = finalizerMethod;
+        this.finalizedResultName = getResultName() + "Result";
```

Comment:
We are still having the problem where there might be a clash. I think what you need to do is 
```
if ( this.finalizerMethod != null ) {
    this.finalizedResultName = Strings.getSafeVariableName( getResultName() + "Result", existingVariableNames )
    existingVariableNames.add( this.finalizedResultName );
}
else {
    this.finalizedResultName = null;
}
```

Reply:
Didn't know about this method. Thank you, I will have a look.